### ISSUE TITLE
Model level annotations should annotate model element rather than model file

### ISSUE THREAD

**Issue by nickerso** [sys-bio/libOmexMeta#78]
Recent discussions have highlighted an issue with the way we are creating "model level" annotations. The current model level annotations generated by libOmexMeta are annotating the model file, but they should be annotating the model contained in the file. For CellML this is the root element of the model file, but for SBML models this is the `model` element which is a child of the `sbml` element. For example, given this python code:

```python
from pyomexmeta import RDF, eUriType

sbml = """<?xml version="1.0" encoding="UTF-8"?>
<sbml xmlns="http://www.sbml.org/sbml/level3/version1/core" level="3" version="1">
  <model metaid="ToyModel" id="ToyModel">
    <listOfCompartments>
      <compartment sboTerm="SBO:0000410" id="cell" metaid="cell" spatialDimensions="3" size="1" constant="true"/>
    </listOfCompartments>
    <listOfSpecies>
      <species id="A" metaid="A" compartment="cell" initialConcentration="10" hasOnlySubstanceUnits="false" boundaryCondition="false" constant="false"/>
      <species id="B" metaid="B" compartment="cell" hasOnlySubstanceUnits="false" boundaryCondition="false" constant="false"/>
    </listOfSpecies>
  </model>
</sbml>
"""

rdf_graph = RDF()
rdf_graph.set_archive_uri("MyOmexArchive.omex")
rdf_graph.set_model_uri("MyModel.sbml")
annot_editor = rdf_graph.to_editor(sbml, generate_new_metaids=False, sbml_semantic_extraction=False)
annot_editor.add_creator("0000-0001-8254-4957") \
    .add_curator("0000-0002-2390-6572") \
    .add_taxon("9895") \
    .add_pubmed("12334") \
    .add_date_created("18-09-2020")

print(rdf_graph)
```

we get the RDF graph:

```ttl
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix pubmed: <https://identifiers.org/pubmed:> .
@prefix dc: <https://dublincore.org/specifications/dublin-core/dcmi-terms/> .
@prefix bqbiol: <http://biomodels.net/biology-qualifiers/> .
@prefix NCBI_Taxon: <https://identifiers.org/taxonomy:> .
@prefix bqmodel: <http://biomodels.net/model-qualifiers/> .
@prefix OMEXlib: <http://omex-library.org/> .
@prefix local: <http://omex-library.org/MyOmexArchive.omex/MyModel.rdf#> .

<http://omex-library.org/MyOmexArchive.omex/MyModel.rdf#>
    dc:creator <https://orcid.org/0000-0002-2390-6572> .

<http://omex-library.org/MyOmexArchive.omex/MyModel.sbml>
    bqbiol:hasTaxon <https://identifiers.org/taxonomy:9895> ;
    bqmodel:isDescribedBy <https://identifiers.org/pubmed:12334> ;
    dc:created [
        dc:W3CDTF "18-09-2020"
    ] ;
    dc:creator <https://orcid.org/0000-0001-8254-4957> .
```

The triples for `<http://omex-library.org/MyOmexArchive.omex/MyModel.sbml>` should be against `<http://omex-library.org/MyOmexArchive.omex/MyModel.sbml#ToyModel>`.

And a comparable example for a CellML model...

```python
cellml = """
<model xmlns="http://www.cellml.org/cellml/1.1#" xmlns:cmeta="http://www.cellml.org/metadata/1.0#"
      name="annotation_examples" cmeta:id="annExamples">
  <component name="main">
  </component>
</model>
"""
rdf_graph = RDF()
rdf_graph.set_archive_uri("MyOmexArchive.omex")
rdf_graph.set_model_uri("MyModel.cellml")
annot_editor = rdf_graph.to_editor(cellml, generate_new_metaids=False, sbml_semantic_extraction=False)
annot_editor.add_creator("0000-0001-8254-4957") \
    .add_curator("0000-0002-2390-6572") \
    .add_taxon("9895") \
    .add_pubmed("12334") \
    .add_date_created("18-09-2020")

print(rdf_graph)
```

results in 

```ttl
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix pubmed: <https://identifiers.org/pubmed:> .
@prefix dc: <https://dublincore.org/specifications/dublin-core/dcmi-terms/> .
@prefix bqbiol: <http://biomodels.net/biology-qualifiers/> .
@prefix NCBI_Taxon: <https://identifiers.org/taxonomy:> .
@prefix bqmodel: <http://biomodels.net/model-qualifiers/> .
@prefix OMEXlib: <http://omex-library.org/> .
@prefix local: <http://omex-library.org/MyOmexArchive.omex/MyModel.rdf#> .

<http://omex-library.org/MyOmexArchive.omex/MyModel.cellml>
    bqbiol:hasTaxon <https://identifiers.org/taxonomy:9895> ;
    bqmodel:isDescribedBy <https://identifiers.org/pubmed:12334> ;
    dc:created [
        dc:W3CDTF "18-09-2020"
    ] ;
    dc:creator <https://orcid.org/0000-0001-8254-4957> .

<http://omex-library.org/MyOmexArchive.omex/MyModel.rdf#>
    dc:creator <https://orcid.org/0000-0002-2390-6572> .
```

and `http://omex-library.org/MyOmexArchive.omex/MyModel.cellml` should be `http://omex-library.org/MyOmexArchive.omex/MyModel.cellml#annExamples`.

For SBML and CellML models, we can likely access the corresponding ID for the `model` element as the default source URI. But it might also be nice to add new API to set/get the `model_id` ? In an SBML model with hierarchical composition, there could be additional models represented in the SBML file (as model definitions, rather than models per se), and it may make sense to make model level annotations for those.

**Comment by matthiaskoenig** [comment]
In addition document level annotations exists on the `<sbml>` element in SBML, i.e. model and document annotations are different in SBML whereas in CellML these are the same annotations. Just mentioning these to raise awareness.

**Comment by CiaranWelsh** [comment]
Implemented in v1.2.10. Do a `pip install pyomexmeta` to see the updates. Any problems, let me know.
